src/Chapter2.hs

```diff
@@ -327,8 +348,13 @@ from it!
 ghci> :l src/Chapter2.hs
 -}
 subList :: Int -> Int -> [a] -> [a]
-subList = error "subList: Not implemented!"
-
+subList x y alist 
+  | length alist == 0 = []
```

Comment:
That is the correct idea!

In Haskell, unfortunately, it is not cheap to get the length of the list. You need to get through each element of the list to get that. So it is not efficient to make this check in here.
However, you can compare this list to `[]` empty list. This operation is lazy, so works instantly:

```haskell
alist == [] = [] 
```

Alternatively, there is `null` function that does precisely the same:

```haskell
null alist = []